src/spot/hanjiSpot.ts

```diff
@@ -407,36 +407,45 @@ export class HanjiSpot implements Disposable {
   }
 
   /**
-   * Retrieves the market information for the specified market.
+   * Retrieves the markets information from cache.
    *
-   * @param {GetMarketParams} params - The parameters for retrieving the market information.
-   * @returns {Promise<Market | undefined>} A Promise that resolves to the market information or undefined if the market is not found.
+   * @returns {Promise<Map<string, Market> | undefined>} A Promise that resolves to the markets information or undefined if error when requesting markets.
    */
-  async getMarket(params: GetMarketParams): Promise<Market | undefined> {
-    let market = this.markets.get(params.market);
+  async getCachedMarkets(): Promise<Map<string, Market> | undefined> {
+    const markets = this.cachedMarkets;
 
-    if (!market) {
+    if (!markets.size) {
       try {
-        let getMarketPromise = this.marketPromises.get(params.market);
-        if (!getMarketPromise) {
-          getMarketPromise = this.getMarkets(params);
-          this.marketPromises.set(params.market, getMarketPromise);
+        let getMarketsPromise = this.cachedMarketsPromise;
+        if (!getMarketsPromise) {
+          getMarketsPromise = this.getMarkets({ market: ALL_MARKETS_ID });
+          this.cachedMarketsPromise = getMarketsPromise;
         }
 
-        const markets = await getMarketPromise;
-        market = markets[0];
-
-        this.marketPromises.delete(params.market);
+        const marketsRes = await getMarketsPromise;
+        this.cachedMarketsPromise = undefined;
+        marketsRes.forEach(market => markets.set(market.id, market));
       }
       catch (error) {
         console.error(error);
       }
-      if (!market)
-        return undefined;
 
-      this.markets.set(params.market, market);
+      if (!markets.size) return undefined;
     }
 
+    return markets;
+  }
+
+  /**
+   * Retrieves the market information for the specified market.
+   *
+   * @param {GetMarketParams} params - The parameters for retrieving the market information.
+   * @returns {Promise<Market | undefined>} A Promise that resolves to the market information or undefined if the market is not found.
+   */
+  async getMarket(params: GetMarketParams): Promise<Market | undefined> {
+    const markets = await this.getMarkets(params);
+    const market = markets[0];
```

Comment:
we should catch error with unknown market or remove this method, `getMarkets` can be used instead